[#1124] Add french localization

pylapp opened the pull request:
<!-- 
Thank you for opening a pull request! 

We assume you have read CONTRIBUTING.md. If you have not, do not be surprised if your PR is rejected for reasons listed therein.

Please note that if your PR does not address an issue that was assigned to you, you are still welcome to open it; however, it will not receive merge priority and may be rejected for scope reasons.
-->

## Issues

- Closes #1124 

## Description

<!-- Describe what this PR does at a high level (e.g., "adds a toggle to do <thing>") -->
Add french support for application.
Translate wordings in french.
Modify a bit Swift code so as to load localized wordings.

## Implementation Notes

<!-- Any information about the code that would be helpful for reviewing -->
You may notice some parts of Swift source code have been modified a bit so as to use localized versions of wordings.
In fact, SwftUI computes the wording to translate at runtime mostly but depending to how you add strings in your views, these strings can be considered as pure string to keep asis, or no localized string key to translated then. The more indirectly you add the string, the string won't get translated. Thus a `localized` method has been defined to force to use the localized version considering the string as wording key. Otherwise it will imply to refactor a lot of views to use the string keys the closest to the views. I use that tricky since eons :-D

pylapp committed:
chore: update project settings to manage FR localz (#1124)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
feat: add french translations (#1124) (WIP)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: use of `LocalizedStringResource` at build time to load suitable wordings (#1124)

Previously some wordings have been declared as LocalizedStringResource,
but this object should be managed at runtime so as to use the suitable bundles and locales to load the expected translated wordings.
Now, with these values as String and only wrapped in views at runtime, the app can use the suitable translations instead of always the default value.

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: add missing wording `Search...` (#1124)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: load the localized version of `Search...` for UI-kit based search bar placeholder (#1124)

The text was never localized, even if wording was transalted, because UIKit layer oes not make the conversion

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: add missing wording for sort types (#1124)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: add missing wording, improve some translations (#1124)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
fix: use localized version of strings (#1124)

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

Sjmarf reviewed (commented):
Nice work :)

It looks like you might have missed translating `"Yes"`?

<img src="https://github.com/user-attachments/assets/bb48731e-b22d-46d2-a48e-cf68b3f688b5" width=50%>

The label for the "Inbox" tab is quite long, which causes the letters to be squished on some devices.

<img src="https://github.com/user-attachments/assets/51c0717d-500e-4094-bb1c-a399ce66b552" width=50%>

You're the expert, of course, but I wonder whether there's a better alternative? It doesn't need to be a direct translation. Reddit translates "Inbox" as "Notifications" in French, for example:

<img src="https://github.com/user-attachments/assets/9113898e-12fd-418b-80a5-a921b6152fb9" width=50%>
<img src="https://github.com/user-attachments/assets/2ce477ae-eae1-422e-a204-25b5954b6a2d" width=50%>

If you think "Boîte de réception" is the best option, I'm happy to stick with that :)

pylapp commented:
Ou that's interesting ideas! I will have a look tomorrow and update the PR with your suggestions.
Let's keep in touch!

Sjmarf commented:
Sounds good :)

I also noticed a few places where we forgot to include some strings in the localization file (oops). 

I've implemented fixes for these on the `sjmarf/localization fixes2` branch of this repository. If you merge that branch into your PR branch, the additional strings will appear in the Localizable file. You can then filter the Localizable file by choosing "State ≠ Translated" in the search bar to see the new strings that need to be translated.

![Screenshot 2025-02-26 at 21 48 57](https://github.com/user-attachments/assets/77cc2f6f-68af-4066-a138-1486d622c281)

You'll probably need to build the app (⌘B) in order for the new strings to appear in the Localizable file.

pylapp committed:
fix: translations for `Yes` and `Inbox` (#1124)

Related to https://github.com/mlemgroup/mlem/pull/1803#pullrequestreview-2645909999

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp committed:
refactor: merge `upstream/sjmarf/localization-fixes2` and add missing translations (#1124)

Related to https://github.com/mlemgroup/mlem/pull/1803#issuecomment-2686284757

Signed-off-by: Pierre-Yves Lapersonne <dev@pylapersonne.info>

pylapp commented:
@Sjmarf Ok, seems good now, I added the fix you mentioned and merged your branch before addind new translations.
I compiled the app ([on my own branch](https://github.com/pylapp/mlem/tree/feat/1124-add-french-l10n)), everything is ok.
The [PR branch](https://github.com/pylapp/mlem/tree/pr/1124-add-french-l10n) is updated :)

pylapp commented:
@Sjmarf Ok, seems good now, I added the fix you mentioned and merged your branch before addind new translations.
I compiled the app ([on my own branch](https://github.com/pylapp/mlem/tree/feat/1124-add-french-l10n)), everything is ok.
The [PR branch](https://github.com/pylapp/mlem/tree/pr/1124-add-french-l10n) is updated :)

Sjmarf reviewed (approved):
Nice, looks good to me! I'm happy for you to merge this pull request, if you've got the option to? If not, let me know and I can merge it. Thanks for contributing to Mlem! :D

pylapp commented:
> Nice, looks good to me! I'm happy for you to merge this pull request, if you've got the option to? If not, let me know and I can merge it. Thanks for contributing to Mlem! :D

Nope I cannot merge this PR as I don’t have *write* access in the repository because I am not part of its team. That’s normal.
Only you and your colleagues are able to do that, I could be nice 😄


![IMG_1451](https://github.com/user-attachments/assets/f579ed3b-fe6a-40d7-9ab8-1cd5f215ce94)

Sjmarf commented:
Thanks, I've merged it 😄 I'd forgotten how it works lol
